rewrite-core/src/main/java/org/openrewrite/config/ClasspathScanningLoader.java

```diff
@@ -98,29 +98,35 @@ public ClasspathScanningLoader(Iterable<Path> compileClasspath, Properties prope
      * @param acceptPackages Limit scan to specified packages
      */
     public ClasspathScanningLoader(String jarName, Iterable<Path> compileClasspath, Properties properties, String[] acceptPackages) {
-        URLClassLoader classpathLoader = new URLClassLoader(
-                stream(compileClasspath.spliterator(), false)
-                        .map(cc -> {
-                            try {
-                                return cc.toUri().toURL();
-                            } catch (MalformedURLException e) {
-                                throw new UncheckedIOException(e);
-                            }
-                        })
-                        .toArray(URL[]::new),
-                getClass().getClassLoader()
-        );
+        this(jarName, compileClasspath, properties, null, acceptPackages);
+    }
+
+    public ClasspathScanningLoader(String jarName, Iterable<Path> compileClasspath, Properties properties, @Nullable ClassLoader classLoader, String[] acceptPackages) {
```

Comment:
Can we move the if condition up into the original constructor so this isn't a breaking change?

Reply:
I tried that first by passing in the generated classloader as a default property. The problem is you can't call `getClass` before or during `this()`.